### PR TITLE
Add pytest

### PR DESCRIPTION
**Pull request by cav71**
This is a simple round-trip test for show-source, not sure if there's something already there.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -2,9 +2,46 @@ name: build
 
 on: [push, pull_request]
 
-
+env:
+  PYTEST_ADDOPTS: "--color=yes"
 
 jobs:
+  test:
```
Comment (justinmayer) [inline .github/workflows/main.yml:10]: Please add an extra line in between the `jobs:` and `test:` lines (makes everything more readable).

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -51,7 +88,7 @@ jobs:
         id: check_release
         run: |
           python -m pip install --upgrade pip
-          python -m pip install poetry githubrelease httpx==0.16.1 autopub
+          python -m pip install poetry githubrelease autopub
```
Comment (justinmayer) [inline .github/workflows/main.yml]: Please revert the change to this line, since `httpx==0.16.1` is needed. (See relevant recent commit for its purpose.)

File: pelican/plugins/test/data/pelicanconf.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*- #
```
Comment (justinmayer) [inline pelican/plugins/test/data/pelicanconf.py]: The `coding` directive is not needed. As of [PEP 3120](https://www.python.org/dev/peps/pep-3120/), the default encoding for Python source is UTF-8. The shebang line can probably also be removed, since this file is not meant to be invoked directly.

File: pelican/plugins/test/data/pelicanconf.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+FEED_ALL_ATOM = None
+CATEGORY_FEED_ATOM = False
+TRANSLATION_FEED_ATOM = None
+AUTHOR_FEED_ATOM = None
+AUTHOR_FEED_RSS = None
+
+# Blogroll
+LINKS = (
+    ("Pelican", "http://getpelican.com/"),
+    ("Python.org", "http://python.org/"),
```
Comment (justinmayer) [inline pelican/plugins/test/data/pelicanconf.py]: These links should start with `https://`

File: pelican/plugins/test/data/pelicanconf.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+)
+
+DEFAULT_PAGINATION = 10
+
+
```
Comment (justinmayer) [inline pelican/plugins/test/data/pelicanconf.py]: Might as well remove one of these empty lines.

**Review by justinmayer — CHANGES_REQUESTED** [review]

**Comment by justinmayer** [comment]
@cav71: Thank you for the submission. I made a few minor suggestions.

@kdeldycke: Would you mind taking a look as well? I only had time to give this a very cursory glance. Thanks!

**Comment by cav71** [comment]
I hope changes are good, thanks

File: pelican/plugins/test/test_show_source.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        "--relative-urls",
+    ]
+    pelican.main([str(a) for a in args])
+    yield output
+
+
+def test_show_source(site):
+    tag = "esse-quam-laboriosam-at-accusantium"
+
+    assert (site / (tag + ".html")).exists()
```
Comment (avaris) [inline pelican/plugins/test/test_show_source.py:30]: Another `exists` check for the source (`.txt`) here?

Comment (cav71) [reply pelican/plugins/test/test_show_source.py:30]: one to test the generated file location and another for the content. Where would be the other exists check?

Comment (avaris) [reply pelican/plugins/test/test_show_source.py:30]: You check that the content has a link to it (below) but you don't check if the source itself is copied over:

```python
assert (site / (tag + ".txt")).exists()
```

Comment (cav71) [reply pelican/plugins/test/test_show_source.py:30]: thanks, I've update the test to include the check for the source file (and put some comment in the code)

Comment (avaris) [reply pelican/plugins/test/test_show_source.py:30]: thanks

**Review by justinmayer — APPROVED** [review]
Many thanks to @cav71 for the enhancement and to @avaris for reviewing. 💫
